my_enumerable.rb

```diff
@@ -0,0 +1,160 @@
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    for value in self
+			yield(value)
+		end
+		return self
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    idx = 0
+    while idx < to_a.length
+      yield(to_a[idx], idx)
+      idx += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    select_arr = []
+    my_each { |value| select_arr << value if yield(value) }
+    select_arr
+  end
+
+  def my_all?(param = nil)
+    if block_given?
+      to_a.my_each { |value| return false if yield(value) == false }
+      return true
+    elsif param.nil?
+      to_a.my_each { |value| return false if value.nil? || value == false }
+    elsif !param.nil? && (param.is_a? Class)
+      to_a.my_each { |value| return false unless [value.class, value.class.superclass].include?(param) }
+    elsif !param.nil? && param.class == Regexp
+      to_a.my_each { |value| return false unless param.match(value) }
+    else
+      to_a.my_each { |value| return false if value != param }
+    end
+    true
+  end
+
+  def my_any?(arg = nil)
+    if block_given?
+      to_a.my_each {|idx| return true if yield(idx)}
+      return false
+      elsif arg.nil?
+        to_a.my_each {|idx| return true if idx}
+      elsif !arg.nil? && (arg.is_a? Class)
+        to_a.my_each {|idx| return true if [idx.class, idx.class.superclass].include?(arg)}
+      elsif !arg.nil? && arg.class == Regexp
+        to_a.my_each {|idx| return true if arg.match(idx)}
+      else
+        to_a.my_each {|idx| return true if idx == arg}
+    end
+    false
+  end
+
+  def my_none?(arg = nil, &block)
+    !my_any?(arg, &block)
+  end
+
+  def my_count
+    count = 0
+    self.my_each { |val| count += 1 if yield(val)}
+    count
+  end
```

Comment:
- [ ] should have working `my_count` **(covered by automated tests)** identical and returns the same thing as ruby's [`count`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-count). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_count.png) from [Odin](https://github.com/TheOdinProject/curriculum/blob/master/ruby_programming/archive/basic_ruby/project_advanced_building_blocks.md#project-2-enumerable-methods))

#### my_count
> Remenber to make it work for ranges and hashes
- [ ] my_count returns the number of items in enum through enumeration
```ruby
[5, 4, 1].my_count == [5, 4, 1].count => 3
```
- my_count is giving a LocalJumpError
![image](https://user-images.githubusercontent.com/52209067/95801103-87642300-0cb6-11eb-9684-e44d36dc627c.png)

- Real count
![image](https://user-images.githubusercontent.com/52209067/77696292-c200ab00-6f72-11ea-8e97-5a2b79256cbc.png)

- [ ] my_count counts the number of items in enum that are equal to the item if an argument is given
```ruby
[5, 4, 1].my_count(4) == [5, 4, 1].count(4) => 1
```
- my_count is giving an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/95801195-d0b47280-0cb6-11eb-93d9-535cd39bd98e.png)

- Real count
![image](https://user-images.githubusercontent.com/52209067/77696378-e2306a00-6f72-11ea-9310-0bb30ab9c94d.png)